File: MarriageCalculator/MarriageCalculator.Core/Models/MarriageGame.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MarriageCalculator.Core.Models;

public class MarriageGame
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    public int Sequence { get; set; }

    [BsonRepresentation(BsonType.ObjectId)]
    public string MarriageGameRoundId { get; set; } = string.Empty;

    [BsonRepresentation(BsonType.ObjectId)]
    public string WinnerId { get; set; } = string.Empty;

    [BsonRepresentation(BsonType.ObjectId)]
    public string DealerId { get; set; } = string.Empty;

    public int TotalMaal { get; set; }
    public bool ClosedRound { get; set; }
    public DateTime CreatedTime { get; set; }

    [BsonIgnore]
    public Dictionary<string, MarriageGameScore> MarriageGameScores { get; set; } = []; //playerId, MarriageGameScore
}

File: MarriageCalculator/Android/app/src/main/java/com/sanjeeb/marriagecalculator/data/model/MarriageGame.kt
package com.sanjeeb.marriagecalculator.data.model

import com.google.gson.annotations.SerializedName

data class MarriageGame(
    @SerializedName("id") val id: String = "",
    @SerializedName("sequence") val sequence: Int = 0,
    @SerializedName("marriageGameRoundId") val marriageGameRoundId: String = "",
    @SerializedName("winnerId") val winnerId: String = "",
    @SerializedName("dealerId") val dealerId: String = "",
    @SerializedName("totalMaal") val totalMaal: Int = 0,
    @SerializedName("closedRound") val closedRound: Boolean = false,
    @SerializedName("createdTime") val createdTime: String = "",
    @SerializedName("marriageGameScores") val marriageGameScores: Map<String, MarriageGameScore>? = null
)

data class CreateMarriageGameRequest(
    @SerializedName("sequence") val sequence: Int,
    @SerializedName("marriageGameRoundId") val marriageGameRoundId: String,
    @SerializedName("winnerId") val winnerId: String,
    @SerializedName("dealerId") val dealerId: String,
    @SerializedName("totalMaal") val totalMaal: Int = 0,
    @SerializedName("closedRound") val closedRound: Boolean = false,
    @SerializedName("scores") val scores: List<CreateScoreRequest> = emptyList()
)

data class CreateScoreRequest(
    @SerializedName("playerId") val playerId: String,
    @SerializedName("seen") val seen: Boolean = false,
    @SerializedName("playing") val playing: Boolean = true,
    @SerializedName("maal") val maal: Int = 0,
    @SerializedName("bonusPoint") val bonusPoint: Int = 0,
    @SerializedName("duply") val duply: Boolean = false,
    @SerializedName("winner") val winner: Boolean = false,
    @SerializedName("deal") val deal: Boolean = false,
    @SerializedName("position") val position: Int = 0
)
